base/uk.ac.stfc.isis.ibex.experimentdetails/src/uk/ac/stfc/isis/ibex/experimentdetails/UserDetails.java

```diff
@@ -85,4 +82,19 @@ public String getAssociatedExperimentStartDate() {
 		return DF.format(associatedExperimentStartDate);
 	}
 
+    /**
+     * Finds the highest role and sets it.
+     * 
+     * @param roleToPrimary
+     *            the role to try to set as primary
+     */
+    public void setPrimaryRole(Role roleToPrimary) {
+        if (roleToPrimary.toString() == "PI") {
```

Comment:
For String comparisons in Java you need to do `roleToPrimary.toString().equals("PI")`, and the same for the other checks below.

You can make this code even nicer by using the enums available in the Role class. For example you should be able to do `roleToPrimary == Role.PI` for the comparison (please double check this works though, I am not 100% sure it does!).